src/core/Global.ttslua

```diff
@@ -619,176 +638,285 @@ end
 -- Content Importing and XML functions
 ---------------------------------------------------------
 
-function onClick_refreshList()
-  local request = WebRequest.get(SOURCE_REPO .. '/library.json', completed_list_update)
-  requestObj = request
-  startLuaCoroutine(Global, 'downloadCoroutine')
+-- forwards the requested content type to the update function and sets highlight to clicked tab
+---@param tabId String Id of the clicked tab
+function onClick_tab(_, _, tabId)
+  for listId, listContent in pairs(tabIdTable) do
+    if listId == tabId then
+      UI.setClass(listId, 'downloadTab activeTab')
+      contentToShow = listContent
+    else
+      UI.setClass(listId, 'downloadTab')
+    end
+  end
+  currentListItem = 1
+  updateDownloadItemList()
 end
 
-function onClick_select(player, params)
-  params = JSON.decode(urldecode(params))
-  local url = SOURCE_REPO .. '/' .. params.url
-  local request = WebRequest.get(url, function (request) complete_obj_download(request, params) end )
-  requestObj = request
+-- click function for the items in the download window
+-- updates backgroundcolor for row panel and fontcolor for list item
+function onClick_select(_, _, identificationKey)
```

Comment:
Unsure what it is, but you have an error case here you're not catching (specifically throwing the error `attempt to concatenate a nil value`).  You can try selecting "Bad Batch" under Fan-Made Player Cards to test

Reply:
Additionally, you might want to consider adding the clicking sound effect to these (but I don't feel strongly about it)